controllers/validator.js

```diff
@@ -0,0 +1,53 @@
+import _ from 'lodash';
+
+function validateObject(obj, schema) {
+
+    let errorUnexpectedMessage;
+    let errorMessage;
+
+    let unexpected = _.difference(Object.keys(obj), Object.keys(schema));
+
+    if (unexpected.length > 0) {
+        errorUnexpectedMessage = unexpected.join(', ') + ' is not defined for the object';
+    }
+
+    let errorString = Object.keys(schema).map(function (key) {
+
+        if (schema[key].required) {
+            if (obj[key] === 'undefined') {
+                errorMessage = key + ' value is required';
+                return errorMessage;
+            }
+        }
+
+        if ((typeof obj[key]) !== schema[key].type) {
+            errorMessage = key + ' type mismatch';
+            return errorMessage;
+        }
+
+        if (schema[key].type === 'object') {
+            let res = validateObject(obj[key], schema[key].properties);
+            return res.errorMessage;  //(*) recursion
+        }
+
+        return;
+    }).filter(function (x) {
+        return x;
+    });
+
+    if (errorUnexpectedMessage) {
+        errorString.unshift(errorUnexpectedMessage);
+    }
+
+    let resObj = {
+        isValid: errorString.length <= 0,
+        errorMessage: errorString.join('; ')
+    };
+
+    return resObj;
+
+}
+
+export default {
```

Comment:
If you use default export, please export this method as a default one.
If you want to export the method as a named export, please don't use default. Please observe:
`export default validateObject;`
vs
`export { validateObject };`